### PR TITLE
COVID immunizations

### PR DESCRIPTION
**Pull request by eanders**

### CODE REVIEW

File: app/models/concerns/obvious_client_matcher.rb [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      incoming_ssn == client_ssn
+    end
+
+    def check_birthday(incoming_dob, client_dob)
+      return false if incoming_dob.blank?
+
+      incoming_dob == client_dob
+    end
+
+    private def check_name(incoming_first, incoming_last, client_first, client_last)
```
Comment (awisema) [inline app/models/concerns/obvious_client_matcher.rb:49]: Why is `check_name` private, when the others aren't?

File: app/models/health/vaccination.rb [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+      # Add new vaccinations
+      system_user_id = User.setup_system_user.id
+      new_vaccinations = []
+      unassigned.preload(:patient).find_each do |vaccination|
+        # Attempt to find the client based on medicaid_id
+        client_id = if vaccination.patient&.client_id.present?
+          vaccination.patient.client_id
+        else
+          all_matches = new.matching_clients(
```
Comment (awisema) [inline app/models/health/vaccination.rb]: I don't really like a stateless new to access what could be a class method...  Move `matching_clients` to `self.matching_clients` and add a method with a `self.class` call?

File: app/models/health/vaccination.rb [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+    def self.use_tsql_import?
+      false
+    end
+
+    private def clean_vaccination_type
+      # TODO: adjust based on real data
+      vaccination_type
+    end
+
+    def follow_up_date
```
Comment (awisema) [inline app/models/health/vaccination.rb:160]: I think this needs the same comment about how it is called.

File: db/health/migrate/20201223182315_create_immunizations.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
+class CreateImmunizations < ActiveRecord::Migration[5.2]
```
Comment (awisema) [inline db/health/migrate/20201223182315_create_immunizations.rb]: Make class name match table name?

File: db/seeds.rb [hunk 5]
```diff
@@ -820,6 +820,15 @@ def report_list
       health: false,
     }
   end
+  if RailsDrivers.loaded.include?(:disability_summary)
```
Comment (awisema) [inline db/seeds.rb]: This should be `:text_message`, yes?

**Review by awisema — CHANGES_REQUESTED** [review]

File: app/models/concerns/obvious_client_matcher.rb (hunk 1 above)
Comment (eanders) [reply app/models/concerns/obvious_client_matcher.rb:49]: All should be private.

File: app/models/health/vaccination.rb (hunk 2 above)
Comment (eanders) [reply app/models/health/vaccination.rb]: The `clients` method caches a lookup table of destination clients.  This was in the wrong place, so I moved it out of the loop and added a note.

File: db/seeds.rb (hunk 5 above)
Comment (eanders) [reply db/seeds.rb]: Yes.

**Review by awisema — APPROVED** [review]

**Pull request merged**